layer/filestore.go

```diff
@@ -66,7 +66,10 @@ func (fms *fileMetadataStore) getMountFilename(mount, filename string) string {
 	return filepath.Join(fms.getMountDirectory(mount), filename)
 }
 
-func (fms *fileMetadataStore) StartTransaction() (*fileMetadataTransaction, error) {
+func (fms *fileMetadataStore) StartTransaction(cacheID string) (*fileMetadataTransaction, error) {
+	if len(cacheID) == 0 {
```

Comment:
It would be nice to have a test for this branch.

Reply:
Done